src/core/ddsi/src/q_xmsg.c

```diff
@@ -1012,6 +1180,109 @@ void nn_xpack_free (struct nn_xpack *xp)
   ddsrt_free (xp);
 }
 
+static ssize_t nn_xpack_send_rtps(struct nn_xpack * xp, const nn_locator_t *loc)
+{
+  ssize_t ret = -1;
+
+#ifdef DDSI_INCLUDE_SECURITY
+  /* Only encode when needed. */
+  if (xp->sec_info.use_rtps_encoding)
```

Comment:
Looks to me like it would be better to refactor this into a function

Reply:
See the new secure_conn_write().